autogpt_platform/backend/backend/blocks/vector_dbs/pinecone.py

```diff
@@ -0,0 +1,127 @@
+from typing import Literal
+
+from autogpt_libs.supabase_integration_credentials_store import APIKeyCredentials
+from pinecone import Pinecone, ServerlessSpec
+
+from backend.data.block import Block, BlockCategory, BlockOutput, BlockSchema
+from backend.data.model import CredentialsField, CredentialsMetaInput, SchemaField
+
+
+class PineconeInitBlock(Block):
+    class Input(BlockSchema):
+        credentials: CredentialsMetaInput[Literal["pinecone"], Literal["api_key"]] = (
+            CredentialsField(
```

Comment:
nit: you can do the same thing you did with jina's _auth here but in file if you want. Not needed but just deduplicates stuff

Reply:
updated